vwifi.c

```diff
@@ -1915,6 +2061,90 @@ static int vwifi_get_tx_power(struct wiphy *wiphy,
     return 0;
 }
 
+/* Make the device join a specific IBSS. */
+static int vwifi_join_ibss(struct wiphy *wiphy,
+                           struct net_device *ndev,
+                           struct cfg80211_ibss_params *params)
+{
+    struct vwifi_vif *vif = ndev_get_vwifi_vif(ndev);
+    /* Validate vif pointer */
+    if (!vif)
+        return -EINVAL;
+    if (mutex_lock_interruptible(&vif->lock))
+        return -ERESTARTSYS;
+    /* Retrieve IBSS configuration parameters */
+    memcpy(vif->ssid, params->ssid, params->ssid_len);
+    vif->ibss_chandef = params->chandef;
+    vif->ssid_len = params->ssid_len;
+    /* When the BSSID is automatically generated by the system, it will not be
+     * passed as a parameter to the join function. */
+    if (params->bssid)
+        memcpy(vif->bssid, params->bssid, ETH_ALEN);
+    else {
+        /* Search for IBSS networks with WPA settings in the IBSS list. If a
+         * matching network exists, join it. Otherwise, create one. */
+        memcpy(vif->bssid, ndev->dev_addr, ETH_ALEN);
+        struct vwifi_vif *ibss_vif = NULL;
+        list_for_each_entry (ibss_vif, &vwifi->ibss_list, ibss_list) {
+            if (!memcmp(ibss_vif->ssid, vif->ssid, vif->ssid_len) &&
```

Comment:
You should check if these two SSID's lengths are matched first. Otherwise the unexpected result might happen:
`ibss_vif->ssid = "test2"`, `vif->ssid = "test"`, `vif->ssid_len = 4`